src/components/navigation/document-tab-panel.tsx

```diff
@@ -0,0 +1,154 @@
+import { inject, observer } from "mobx-react";
+import React from "react";
+import { BaseComponent, IBaseProps } from "../base";
+import { Tab, Tabs, TabList, TabPanel } from "react-tabs";
+import { ELeftTabSectionType, LeftTabSpec, LeftTabSectionModelType } from "../../models/view/left-tabs";
+import { IStores } from "../../models/stores/stores";
+import { TabPanelDocumentsSection } from "../thumbnail/tab-panel-documents-section";
+import { DocumentDragKey, DocumentModelType, SupportPublication } from "../../models/document/document";
+import { LogEventName, Logger } from "../../lib/logger";
+import { EditableDocumentContent } from "../document/editable-document-content";
+
+import "./document-tab-panel.sass";
+
+interface IProps extends IBaseProps {
+  tabSpec: LeftTabSpec;
+  onDocumentClick?: (document: DocumentModelType) => void;
```

Comment:
Ultimately, I think we should remove all default side effects/behaviors from this component and have them passed in as props rather than have some of them defaulted and some of them passed in. It doesn't need to happen all at once, however, so this is fine as a step in the right direction.

Reply:
Yes, I entirely agree.  I even left in the `handleDocumentClick` for now, but I'm assuming it, and the rest, can come out once we make more progress.